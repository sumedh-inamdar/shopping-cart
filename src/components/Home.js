import React from 'react';
import homeBG from '../assets/homepage-bg-red.jpg';
import rackets from '../assets/featured collections/rackets.jpg';
import mensClothing from '../assets/featured collections/mens-clothing.jpg';
import womensClothing from '../assets/featured collections/womens-clothing.jpg';
import strings from '../assets/featured collections/strings.jpg';
import serve from '../assets/homepage-serve.jpg';
import style from '../assets/homepage-style.jpg';
import solar from '../assets/homepage-solar.jpg';
import variety from '../assets/homepage-variety.jpg';
import robot from '../assets/homepage-robot.jpg';
import { Link } from 'react-router-dom';
import { useInView } from 'react-intersection-observer';

// begins animation after assets are loaded
function showAnimation(event) {
  event.target.closest('.animate-hold').classList.remove('animate-hold');
}

function HomeSectionOne() {
  const { ref, inView } = useInView({
    triggerOnce: true,
    threshold: 0.35
  });

  return (
    <div
      className="max-w-6xl py-12 md:py-32 px-6 mx-auto min-h-[350px]"
      ref={ref}>
      {inView ? (
        <div>
          <div className="text-4xl md:text-5xl mb-6 text-center animate-fade-in-top-0">
            Featured Collections
          </div>
          <ul className="text-center text-2xl font-light flex flex-col md:flex-row justify-around space-y-6 md:space-y-0 md:space-x-6">
            <Link to="/store/racquets/">
              <li className="space-y-4 cursor-pointer animate-fade-in-top-250 opacity-0 animate-hold">
                <div className="hover:scale-105 transition-transform hover:text-red-500 duration-200">
                  <img
                    onLoad={showAnimation}
                    src={rackets}
                    className="m-auto w-64 shadow-xl"
                  />
                  <p className="">Tennis Rackets</p>
                </div>
              </li>
            </Link>
            <Link to="/store/mensClothing/">
              <li className="space-y-4 cursor-pointer animate-fade-in-top-500 opacity-0 animate-hold">
                <div className="hover:scale-105 transition-transform hover:text-red-500 duration-200">
                  <img
                    onLoad={showAnimation}
                    src={mensClothing}
                    className="m-auto w-64 shadow-xl"
                  />
                  <p className="">Men{`'`}s Clothing</p>
                </div>
              </li>
            </Link>
            <Link to="/store/womensClothing/">
              <li className="space-y-4 cursor-pointer animate-fade-in-top-750 opacity-0 animate-hold">
                <div className="hover:scale-105 transition-transform hover:text-red-500 duration-200">
                  <img
                    onLoad={showAnimation}
                    src={womensClothing}
                    className="m-auto w-64 shadow-xl"
                  />
                  <p className="">Women{`'`}s Clothing</p>
                </div>
              </li>
            </Link>
            <Link to="/store/strings/">
              <li className="space-y-4 cursor-pointer animate-fade-in-top-1000 opacity-0 animate-hold">
                <div className="hover:scale-105 transition-transform hover:text-red-500 duration-200">
                  <img
                    onLoad={showAnimation}
                    src={strings}
                    className="m-auto w-64 shadow-xl"
                  />
                  <p className="">Strings</p>
                </div>
              </li>
            </Link>
          </ul>
        </div>
      ) : undefined}
    </div>
  );
}
function HomeSectionTwo() {
  return (
    <div className="bg-slate-200 py-12 md:py-32 px-6">
      <div className="max-w-6xl mx-auto min-h-[550px] space-y-16">
        <div className="flex flex-col md:flex-row w-full space-x-0 md:space-x-6">
          <div className="flex-1 mb-6 md:mb-0">
            <img className="shadow-xl" src={serve} alt="person serving" />
          </div>
          <div className="flex-1 space-y-6 h-min m-auto">
            <p className="text-4xl md:text-5xl">Elevate your game</p>
            <div className="text-lg md:text-xl text-justify">
              Step in and enable your finest game with the right gear. Our
              trained staff can help your pick the right racket, strings, or
              answer any questions.
            </div>
          </div>
        </div>
        <div className="flex flex-col md:flex-row w-full space-x-0 md:space-x-6">
          <div className="flex-1 space-y-6 h-min m-auto">
            <p className="text-4xl md:text-5xl md:text-right">Win with style</p>
            <div className="text-lg md:text-xl md:text-right">
              Feel amazing by winning in style with our fresh apparel collection
              from Nike, Adidas, and more!
            </div>
          </div>
          <div className="flex-1 order-first md:order-last mb-6 md:mb-0">
            <img
              className="shadow-xl"
              src={style}
              alt="girl sitting on court"
            />
          </div>
        </div>
      </div>
    </div>
  );
}
function HomeSectionThree() {
  const { ref, inView } = useInView({
    triggerOnce: true,
    threshold: 0.5
  });
  return (
    <div className="bg-white min-h-[350px] py-12 md:py-32 px-6" ref={ref}>
      {inView ? (
        <div className="max-w-6xl mx-auto space-y-10 md:space-y-16">
          <div className="text-4xl md:text-5xl text-center animate-fade-in-top-0">
            Why You{"'"}ll Love TENNSCENTRL
          </div>
          <ul className="flex flex-col md:flex-row justify-between space-y-10 md:space-y-0 md:space-x-6">
            <li className="flex-1 opacity-0 animate-fade-in-top-250">
              <img
                className="h-52 rounded-full m-auto"
                src={solar}
                alt="solar planels"
              />
              <div className="text-center text-lg mt-5">Sustainability</div>
              <div className="text-center text-medium font-light mt-5">
                Laborum consequat aliquip et in dolore enim veniam laboris
                adipisicing.Non veniam fugiat culpa proident enim in esse.
              </div>
            </li>
            <li className="flex-1 opacity-0 animate-fade-in-top-500">
              <img
                className="h-52 rounded-full m-auto"
                src={variety}
                alt="solar planels"
              />
              <div className="text-center text-lg mt-5">Variety</div>
              <div className="text-center text-medium font-light mt-5">
                Eu deserunt cillum aliqua sint velit est et ad eu eiusmod
                aliquip aliqua ex.Consequat cillum commodo dolor ipsum fugiat in
                excepteur id sunt.
              </div>
            </li>
            <li className="flex-1 opacity-0 animate-fade-in-top-750">
              <img
                className="h-52 rounded-full m-auto"
                src={robot}
                alt="factory robot"
              />
              <div className="text-center text-lg mt-5">Innovation</div>
              <div className="text-center text-medium font-light mt-5">
                Excepteur cupidatat anim mollit excepteur ut quis.Elit nisi
                Lorem aliqua enim nostrud fugiat ex aliquip nostrud ullamco
                excepteur aute amet.
              </div>
            </li>
          </ul>
        </div>
      ) : undefined}
    </div>
  );
}
export default function Home() {
  return (
    <div id="home">
      <div className="bg-gradient-to-r from-teal-500 to-indigo-500">
        <div className="h-[540px] md:h-[720px] max-w-6xl mx-auto flex ">
          <div className="flex-1 h-full">
            <div className="flex flex-col h-full justify-center mx-8 md:mx-0 relative md:left-1/4 md:w-3/4 space-y-7 text-white">
              <div className="flex flex-col text-6xl space-y-3">
                <div className="opacity-0 animate-fade-in-left-250">GAME</div>
                <div className="opacity-0 animate-fade-in-left-500">SET</div>
                <div className="opacity-0 animate-fade-in-left-750">MATCH</div>
              </div>
              <div className="text-xl opacity-0 animate-fade-in-top-1000">
                Your number one premier tennis equipment supplier.
              </div>
              <Link
                to="/store"
                className="opacity-0 animate-fade-in-top-1000 md:w-3/4 bg-blue-400 hover:bg-blue-500 transition-colors pt-1 border-2">
                <button className="w-full">Shop Now</button>
              </Link>
            </div>
          </div>
          <div className="hidden md:block flex-1 m-auto mx-6">
            <img
              onLoad={showAnimation}
              className="border-2 object-scale-down rounded-lg drop-shadow-lg opacity-0 animate-fade-in-top-1000 animate-hold"
              src={homeBG}
            />
          </div>
        </div>
      </div>
      <HomeSectionOne />
      <HomeSectionTwo />
      <HomeSectionThree />
    </div>
  );
}
