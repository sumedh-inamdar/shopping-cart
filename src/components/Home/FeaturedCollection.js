import React from 'react';
import PropTypes from 'prop-types';

export default function FeaturedCollection({
  animation,
  showAnimation,
  image,
  title
}) {
  return (
    <li
      className={`${animation} space-y-4 cursor-pointer opacity-0 animate-hold`}>
      <div className="hover:scale-105 transition-transform hover:text-red-500 duration-200">
        <img
          onLoad={showAnimation}
          src={image}
          className="m-auto w-64 shadow-xl"
        />
        <p>{title}</p>
      </div>
    </li>
  );
}
FeaturedCollection.propTypes = {
  animation: PropTypes.string,
  showAnimation: PropTypes.func,
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
};
