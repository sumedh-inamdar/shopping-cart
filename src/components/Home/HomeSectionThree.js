import React from 'react';
import solar from '../../assets/homepage-solar.jpg';
import variety from '../../assets/homepage-variety.jpg';
import robot from '../../assets/homepage-robot.jpg';
import { useInView } from 'react-intersection-observer';

export default function HomeSectionThree() {
  const { ref, inView } = useInView({
    triggerOnce: true,
    threshold: 0.5
  });
  return (
    <div className="bg-white min-h-[350px] py-12 md:py-32 px-6" ref={ref}>
      {inView ? (
        <div className="max-w-6xl mx-auto space-y-10 md:space-y-16">
          <div className="text-4xl md:text-5xl text-center animate-fade-in-top-0">
            {`Why You'll Love TENNSCENTRL`}
          </div>
          <ul className="flex flex-col md:flex-row justify-between space-y-10 md:space-y-0 md:space-x-6">
            <li className="flex-1 opacity-0 animate-fade-in-top-250">
              <img
                className="h-52 rounded-full m-auto"
                src={solar}
                alt="solar planels"
              />
              <div className="text-center text-lg mt-5">Sustainability</div>
              <div className="text-center text-medium font-light mt-5">
                Laborum consequat aliquip et in dolore enim veniam laboris
                adipisicing.Non veniam fugiat culpa proident enim in esse.
              </div>
            </li>
            <li className="flex-1 opacity-0 animate-fade-in-top-500">
              <img
                className="h-52 rounded-full m-auto"
                src={variety}
                alt="solar planels"
              />
              <div className="text-center text-lg mt-5">Variety</div>
              <div className="text-center text-medium font-light mt-5">
                Eu deserunt cillum aliqua sint velit est et ad eu eiusmod
                aliquip aliqua ex.Consequat cillum commodo dolor ipsum fugiat in
                excepteur id sunt.
              </div>
            </li>
            <li className="flex-1 opacity-0 animate-fade-in-top-750">
              <img
                className="h-52 rounded-full m-auto"
                src={robot}
                alt="factory robot"
              />
              <div className="text-center text-lg mt-5">Innovation</div>
              <div className="text-center text-medium font-light mt-5">
                Excepteur cupidatat anim mollit excepteur ut quis.Elit nisi
                Lorem aliqua enim nostrud fugiat ex aliquip nostrud ullamco
                excepteur aute amet.
              </div>
            </li>
          </ul>
        </div>
      ) : undefined}
    </div>
  );
}
