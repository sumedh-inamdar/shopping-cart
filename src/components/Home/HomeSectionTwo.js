import React from 'react';
import serve from '../../assets/homepage-serve.jpg';
import style from '../../assets/homepage-style.jpg';

export default function HomeSectionTwo() {
  return (
    <div className="bg-slate-100 py-12 md:py-32 px-6">
      <div className="max-w-6xl mx-auto min-h-[550px] space-y-16">
        <div className="flex flex-col md:flex-row w-full space-x-0 md:space-x-6">
          <div className="flex-1 mb-6 md:mb-0">
            <img className="shadow-xl" src={serve} alt="person serving" />
          </div>
          <div className="flex-1 space-y-6 h-min m-auto">
            <p className="text-4xl md:text-5xl">Elevate your game</p>
            <div className="text-lg md:text-xl text-justify">
              Step in and enable your finest game with the right gear. Our
              trained staff can help your pick the right racket, strings, or
              answer any questions.
            </div>
          </div>
        </div>
        <div className="flex flex-col md:flex-row w-full space-x-0 md:space-x-6">
          <div className="flex-1 space-y-6 h-min m-auto">
            <p className="text-4xl md:text-5xl md:text-right">Win with style</p>
            <div className="text-lg md:text-xl md:text-right">
              Feel amazing by winning in style with our fresh apparel collection
              from Nike, Adidas, and more!
            </div>
          </div>
          <div className="flex-1 order-first md:order-last mb-6 md:mb-0">
            <img
              className="shadow-xl"
              src={style}
              alt="girl sitting on court"
            />
          </div>
        </div>
      </div>
    </div>
  );
}
