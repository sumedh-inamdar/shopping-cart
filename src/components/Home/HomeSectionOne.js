import React from 'react';
import rackets from '../../assets/featured collections/rackets.jpg';
import mensClothing from '../../assets/featured collections/mens-clothing.jpg';
import womensClothing from '../../assets/featured collections/womens-clothing.jpg';
import strings from '../../assets/featured collections/strings.jpg';
import { useInView } from 'react-intersection-observer';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import FeaturedCollection from './FeaturedCollection';

export default function HomeSectionOne({ showAnimation }) {
  const { ref, inView } = useInView({
    triggerOnce: true,
    threshold: 0.35
  });

  return (
    <div
      className="max-w-6xl py-12 md:py-32 px-6 mx-auto min-h-[350px]"
      ref={ref}>
      {inView ? (
        <div>
          <div className="text-4xl md:text-5xl mb-6 text-center animate-fade-in-top-0">
            Featured Collections
          </div>
          <ul className="text-center text-2xl font-light flex flex-col md:flex-row justify-around space-y-6 md:space-y-0 md:space-x-6">
            <Link to="/store/racquets/">
              <FeaturedCollection
                animation={`animate-fade-in-top-250`}
                showAnimation={showAnimation}
                image={rackets}
                title={`Tennis Racquets`}
              />
            </Link>
            <Link to="/store/mensClothing/">
              <FeaturedCollection
                animation={`animate-fade-in-top-500`}
                showAnimation={showAnimation}
                image={mensClothing}
                title={`Men's Clothing`}
              />
            </Link>
            <Link to="/store/womensClothing/">
              <FeaturedCollection
                animation={`animate-fade-in-top-750`}
                showAnimation={showAnimation}
                image={womensClothing}
                title={`Women's Clothing`}
              />
            </Link>
            <Link to="/store/strings/">
              <FeaturedCollection
                animation={`animate-fade-in-top-1000`}
                showAnimation={showAnimation}
                image={strings}
                title={`Strings`}
              />
            </Link>
          </ul>
        </div>
      ) : undefined}
    </div>
  );
}
HomeSectionOne.propTypes = {
  showAnimation: PropTypes.func
};
