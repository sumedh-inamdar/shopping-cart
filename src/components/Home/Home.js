import React from 'react';
import homeBG from '../../assets/homepage-bg-red.jpg';
import HomeSectionOne from './HomeSectionOne';
import HomeSectionTwo from './HomeSectionTwo';
import HomeSectionThree from './HomeSectionThree';
import { Link } from 'react-router-dom';

// begins animation after assets are loaded
function showAnimation(event) {
  event.target.closest('.animate-hold').classList.remove('animate-hold');
}

export default function Home() {
  return (
    <div id="home">
      <div className="bg-gradient-to-r from-teal-500 to-indigo-500">
        <div className="h-[540px] md:h-[720px] max-w-6xl mx-auto flex px-6">
          <div className="flex-1 h-full">
            <div className="flex flex-col h-full justify-center mx-8 md:mx-0 relative md:left-1/4 md:w-3/4 space-y-7 text-white">
              <div className="flex flex-col text-6xl space-y-3">
                <div className="opacity-0 animate-fade-in-left-250">GAME</div>
                <div className="opacity-0 animate-fade-in-left-500">SET</div>
                <div className="opacity-0 animate-fade-in-left-750">MATCH</div>
              </div>
              <div className="text-xl opacity-0 animate-fade-in-top-1000">
                Your number one premier tennis equipment supplier.
              </div>
              <Link
                to="/store/racquets"
                className="opacity-0 animate-fade-in-top-1000 md:w-3/4 bg-blue-400 hover:bg-blue-500 transition-colors pt-1 border-2">
                <button className="w-full">Shop Now</button>
              </Link>
            </div>
          </div>
          <div className="hidden md:block flex-1 m-auto mx-6">
            <img
              onLoad={showAnimation}
              className="border-2 object-scale-down rounded-lg drop-shadow-lg opacity-0 animate-fade-in-top-1000 animate-hold"
              src={homeBG}
            />
          </div>
        </div>
      </div>
      <HomeSectionOne showAnimation={showAnimation} />
      <HomeSectionTwo />
      <HomeSectionThree />
    </div>
  );
}
