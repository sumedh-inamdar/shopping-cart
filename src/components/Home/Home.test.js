/* eslint-disable unicorn/no-null */
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Home from './Home';
import { mockAllIsIntersecting } from 'react-intersection-observer/test-utils';

test('snapshot', () => {
  const { container } = render(<Home />, { wrapper: BrowserRouter });
  expect(container).toMatchSnapshot();
});

describe('intersection observer', () => {
  it('render items upon view', () => {
    render(<Home />, { wrapper: BrowserRouter });

    expect(screen.queryByText('Featured Collections')).toBeNull();
    expect(screen.queryByText(`Why You'll Love TENNSCENTRL`)).toBeNull();
    mockAllIsIntersecting(true);
    expect(screen.queryByText('Featured Collections')).not.toBeNull();
    expect(screen.queryByText(`Why You'll Love TENNSCENTRL`)).not.toBeNull();
  });
});
