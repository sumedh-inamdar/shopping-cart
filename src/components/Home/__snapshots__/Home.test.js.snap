// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <div
    id="home"
  >
    <div
      class="bg-gradient-to-r from-teal-500 to-indigo-500"
    >
      <div
        class="h-[540px] md:h-[720px] max-w-6xl mx-auto flex px-6"
      >
        <div
          class="flex-1 h-full"
        >
          <div
            class="flex flex-col h-full justify-center mx-8 md:mx-0 relative md:left-1/4 md:w-3/4 space-y-7 text-white"
          >
            <div
              class="flex flex-col text-6xl space-y-3"
            >
              <div
                class="opacity-0 animate-fade-in-left-250"
              >
                GAME
              </div>
              <div
                class="opacity-0 animate-fade-in-left-500"
              >
                SET
              </div>
              <div
                class="opacity-0 animate-fade-in-left-750"
              >
                MATCH
              </div>
            </div>
            <div
              class="text-xl opacity-0 animate-fade-in-top-1000"
            >
              Your number one premier tennis equipment supplier.
            </div>
            <a
              class="opacity-0 animate-fade-in-top-1000 md:w-3/4 bg-blue-400 hover:bg-blue-500 transition-colors pt-1 border-2"
              href="/store/racquets"
            >
              <button
                class="w-full"
              >
                Shop Now
              </button>
            </a>
          </div>
        </div>
        <div
          class="hidden md:block flex-1 m-auto mx-6"
        >
          <img
            class="border-2 object-scale-down rounded-lg drop-shadow-lg opacity-0 animate-fade-in-top-1000 animate-hold"
            src="homepage-bg-red.jpg"
          />
        </div>
      </div>
    </div>
    <div
      class="max-w-6xl py-12 md:py-32 px-6 mx-auto min-h-[350px]"
    />
    <div
      class="bg-slate-100 py-12 md:py-32 px-6"
    >
      <div
        class="max-w-6xl mx-auto min-h-[550px] space-y-16"
      >
        <div
          class="flex flex-col md:flex-row w-full space-x-0 md:space-x-6"
        >
          <div
            class="flex-1 mb-6 md:mb-0"
          >
            <img
              alt="person serving"
              class="shadow-xl"
              src="homepage-serve.jpg"
            />
          </div>
          <div
            class="flex-1 space-y-6 h-min m-auto"
          >
            <p
              class="text-4xl md:text-5xl"
            >
              Elevate your game
            </p>
            <div
              class="text-lg md:text-xl text-justify"
            >
              Step in and enable your finest game with the right gear. Our trained staff can help your pick the right racket, strings, or answer any questions.
            </div>
          </div>
        </div>
        <div
          class="flex flex-col md:flex-row w-full space-x-0 md:space-x-6"
        >
          <div
            class="flex-1 space-y-6 h-min m-auto"
          >
            <p
              class="text-4xl md:text-5xl md:text-right"
            >
              Win with style
            </p>
            <div
              class="text-lg md:text-xl md:text-right"
            >
              Feel amazing by winning in style with our fresh apparel collection from Nike, Adidas, and more!
            </div>
          </div>
          <div
            class="flex-1 order-first md:order-last mb-6 md:mb-0"
          >
            <img
              alt="girl sitting on court"
              class="shadow-xl"
              src="homepage-style.jpg"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="bg-white min-h-[350px] py-12 md:py-32 px-6"
    />
  </div>
</div>
`;
